engine/class_modules/sc_death_knight.cpp

```diff
@@ -3374,6 +3387,80 @@ struct auto_attack_t : public death_knight_melee_attack_t
 // Death Knight Abilities
 // ==========================================================================
 
+// Abomination Limb =========================================================
+
+struct abomination_limb_damage_t : public death_knight_spell_t
+{
+  abomination_limb_damage_t( death_knight_t* p ) :
+    death_knight_spell_t( "abomination_limb_damage", p, p -> covenant.abomination_limb -> effectN( 2 ).trigger() )
+  {
+    background = true;
+    base_multiplier *= 1.0 + p -> conduits.brutal_grasp.percent();
+  }
+};
+
+struct abomination_limb_buff_t : public buff_t
+{
+  abomination_limb_damage_t* damage; // (AOE) damage that ticks every second
+
+  abomination_limb_buff_t( death_knight_t* p ) :
+    buff_t( p, "abomination_limb", p -> covenant.abomination_limb ),
+    damage( new abomination_limb_damage_t( p ) )
+  {
+    cooldown -> duration = 0_ms; // Controlled by the action
+    set_tick_callback( [ this ]( buff_t* /* buff */, int /* total_ticks */, timespan_t /* tick_time */ )
+    {
+      damage -> execute();
+    } );
+    set_partial_tick( true );
+  }
+};
+
+struct abomination_limb_t : public death_knight_spell_t
+{
+  int bone_shield_stack_gain;
+  abomination_limb_t( death_knight_t* p, const std::string& options_str ) :
+    death_knight_spell_t( "abomination_limb", p, p -> covenant.abomination_limb )
+  {
+    may_crit = may_miss = may_dodge = may_parry = false;
+
+    parse_options( options_str );
+
+    // Periodic behavior handled by the buff
+    dot_duration = base_tick_time = 0_ms;
+
+    bone_shield_stack_gain = as<int>(p -> covenant.abomination_limb -> effectN( 3 ).base_value());
+
+    if ( action_t* abomination_limb_damage = p -> find_action( "abomination_limb_damage" ) )
+    {
+      add_child( abomination_limb_damage );
+    }
+  }
+
+  void execute() override
+  {
+    death_knight_spell_t::execute();
+
+    p() -> buffs.abomination_limb -> trigger();
+
+    // Through testing we find that
```

Comment:
? 
Might use a comment to explain that abom limb's implementation is simplified to trigger the gains on cast rather than the first time it deals damage as the tooltip says.
You should also mention that the pull effect is NYI